In [2]:
import pandas as pd
import numpy as np

## Import data

In [3]:
# read in data and print the dimensions 2 times to be careful
all_data_df = pd.read_csv('all_data_df.csv')
print(all_data_df.shape)
all_data_df

(29902, 674)


/Users/flowerbed/Desktop/School/9th Grade 2020 - 2021/2021 UCSF AI4ALL/GitHub/ai4all_nhanes/env/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3441: DtypeWarning: Columns (600,601) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


SEQN  ACD011A  ACD011B  ACD011C  ACD040  ACD110  ALQ101  ALQ110  \
0      62161.0      1.0      NaN      NaN     NaN     NaN     2.0     2.0   
1      62163.0      NaN      NaN      NaN     NaN     5.0     NaN     NaN   
2      62164.0      1.0      NaN      NaN     NaN     NaN     NaN     NaN   
3      62165.0      1.0      NaN      NaN     NaN     NaN     NaN     NaN   
4      62169.0      NaN      NaN      NaN     NaN     5.0     1.0     NaN   
...        ...      ...      ...      ...     ...     ...     ...     ...   
29897  93635.0      NaN      NaN      NaN     NaN     NaN     NaN     NaN   
29898  93678.0      NaN      NaN      NaN     NaN     NaN     NaN     NaN   
29899  93688.0      NaN      NaN      NaN     NaN     NaN     NaN     NaN   
29900  93692.0      NaN      NaN      NaN     NaN     NaN     NaN     NaN   
29901  93698.0      NaN      NaN      NaN     NaN     NaN     NaN     NaN   

       ALQ120Q  ALQ120U  ...  WHD080Q  WHD080R  WHD080S  WHD080T  WHD080L  \
0          NaN      NaN  ...      NaN      NaN      NaN      NaN      NaN   
1          NaN      NaN  ...      NaN      NaN      NaN      NaN      NaN   
2          NaN      NaN  ...     43.0      NaN      NaN      NaN      NaN   
3          NaN      NaN  ...      NaN      NaN      NaN      NaN      NaN   
4          1.0      2.0  ...      NaN      NaN      NaN      NaN      NaN   
...        ...      ...  ...      ...      ...      ...      ...      ...   
29897      NaN      NaN  ...      NaN      NaN      NaN      NaN      NaN   
29898      NaN      NaN  ...      NaN      NaN      NaN      NaN      NaN   
29899      NaN      NaN  ...      NaN      NaN      NaN      NaN      NaN   
29900      NaN      NaN  ...      NaN      NaN      NaN      NaN      NaN   
29901      NaN      NaN  ...      NaN      NaN      NaN      NaN      NaN   

       WHD110  WHD120  WHD130  WHD140  WHQ150  
0         NaN     NaN     NaN     NaN     NaN  
1         NaN     NaN     NaN     NaN     NaN  
2       145.0   118.0     NaN   172.0    38.0  
3         NaN     NaN     NaN     NaN     NaN  
4         NaN     NaN     NaN   125.0     NaN  
...       ...     ...     ...     ...     ...  
29897     NaN     NaN     NaN     NaN     NaN  
29898     NaN     NaN     NaN     NaN     NaN  
29899     NaN     NaN     NaN     NaN     NaN  
29900     NaN     NaN     NaN     NaN     NaN  
29901     NaN     NaN     NaN     NaN     NaN  

[29902 rows x 674 columns]

In [4]:
# get column names because why not?
all_data_df.columns

Index(['SEQN', 'ACD011A', 'ACD011B', 'ACD011C', 'ACD040', 'ACD110', 'ALQ101',
       'ALQ110', 'ALQ120Q', 'ALQ120U',
       ...
       'WHD080Q', 'WHD080R', 'WHD080S', 'WHD080T', 'WHD080L', 'WHD110',
       'WHD120', 'WHD130', 'WHD140', 'WHQ150'],
      dtype='object', length=674)

## Categorize values
### Reference Chart
Table 2 @ https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1495268/

### Format mental health data

In [5]:
# get the mental health questions from the master dataframe
mental_health_df = all_data_df.loc[:, 'DPQ010':'DPQ100']
mental_health_df

DPQ010        DPQ020        DPQ030        DPQ040        DPQ050  \
0      5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
1               NaN           NaN           NaN           NaN           NaN   
2               NaN           NaN           NaN           NaN           NaN   
3               NaN           NaN           NaN           NaN           NaN   
4      1.000000e+00  5.397605e-79  1.000000e+00  1.000000e+00  1.000000e+00   
...             ...           ...           ...           ...           ...   
29897           NaN           NaN           NaN           NaN           NaN   
29898           NaN           NaN           NaN           NaN           NaN   
29899           NaN           NaN           NaN           NaN           NaN   
29900           NaN           NaN           NaN           NaN           NaN   
29901           NaN           NaN           NaN           NaN           NaN   

             DPQ060        DPQ070        DPQ080        DPQ090        DPQ100  
0      5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79           NaN  
1               NaN           NaN           NaN           NaN           NaN  
2               NaN           NaN           NaN           NaN           NaN  
3               NaN           NaN           NaN           NaN           NaN  
4      2.000000e+00  5.397605e-79  1.000000e+00  5.397605e-79  5.397605e-79  
...             ...           ...           ...           ...           ...  
29897           NaN           NaN           NaN           NaN           NaN  
29898           NaN           NaN           NaN           NaN           NaN  
29899           NaN           NaN           NaN           NaN           NaN  
29900           NaN           NaN           NaN           NaN           NaN  
29901           NaN           NaN           NaN           NaN           NaN  

[29902 rows x 10 columns]

In [6]:
# pre-check
mental_health_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29902 entries, 0 to 29901
Data columns (total 10 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   DPQ010  15513 non-null  float64
 1   DPQ020  15509 non-null  float64
 2   DPQ030  15507 non-null  float64
 3   DPQ040  15504 non-null  float64
 4   DPQ050  15504 non-null  float64
 5   DPQ060  15502 non-null  float64
 6   DPQ070  15501 non-null  float64
 7   DPQ080  15500 non-null  float64
 8   DPQ090  15498 non-null  float64
 9   DPQ100  10452 non-null  float64
dtypes: float64(10)
memory usage: 2.3 MB


In [7]:
# drop all rows that are completely empty
mental_health_df = mental_health_df.dropna(how='all')
mental_health_df

DPQ010        DPQ020        DPQ030        DPQ040        DPQ050  \
0      5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
4      1.000000e+00  5.397605e-79  1.000000e+00  1.000000e+00  1.000000e+00   
7      1.000000e+00  2.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00   
8      5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
9      5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
...             ...           ...           ...           ...           ...   
28912  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
28915  5.397605e-79  1.000000e+00  5.397605e-79  5.397605e-79  5.397605e-79   
28916  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
28917  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
28921  5.397605e-79  5.397605e-79  1.000000e+00  1.000000e+00  5.397605e-79   

             DPQ060        DPQ070        DPQ080        DPQ090        DPQ100  
0      5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79           NaN  
4      2.000000e+00  5.397605e-79  1.000000e+00  5.397605e-79  5.397605e-79  
7      5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  1.000000e+00  
8      5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79           NaN  
9      5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79           NaN  
...             ...           ...           ...           ...           ...  
28912  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79           NaN  
28915  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  2.000000e+00  
28916  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79           NaN  
28917  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79           NaN  
28921  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  

[15513 rows x 10 columns]

In [8]:
# post-check to make sure rows are dropped
mental_health_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 15513 entries, 0 to 28921
Data columns (total 10 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   DPQ010  15513 non-null  float64
 1   DPQ020  15509 non-null  float64
 2   DPQ030  15507 non-null  float64
 3   DPQ040  15504 non-null  float64
 4   DPQ050  15504 non-null  float64
 5   DPQ060  15502 non-null  float64
 6   DPQ070  15501 non-null  float64
 7   DPQ080  15500 non-null  float64
 8   DPQ090  15498 non-null  float64
 9   DPQ100  10452 non-null  float64
dtypes: float64(10)
memory usage: 1.3 MB


**Question:** what does the 5.397605e-79 value mean? Does it represent the value 5 or a missing value?

We can answer this question by 
- compiling all the data to a question (represented by a column),
- getting the count of each value in the data,
- and comparing it to the count [here](https://wwwn.cdc.gov/Nchs/Nhanes/2011-2012/DPQ_G.htm#Data_Processing_and_Editing).

In [9]:
# compile data
# let's check with DPQ010
mental_health_df.groupby(['DPQ010']).size()

DPQ010
5.397605e-79    11507
1.000000e+00     2533
2.000000e+00      762
3.000000e+00      682
7.000000e+00       10
9.000000e+00       19
dtype: int64

In [10]:
# let's check it again with a different way because it never hurts to be extra cautious
mental_health_df['DPQ010'].value_counts()

5.397605e-79    11507
1.000000e+00     2533
2.000000e+00      762
3.000000e+00      682
9.000000e+00       19
7.000000e+00       10
Name: DPQ010, dtype: int64

Oh no! The count doesn't match up! This is because the data I'm using is from 2011-2016 while the data the chart has is from 2011-2012. Let's add up all the counts from 2011 to 2016 manually since I don't want to do it the other way.

**0:** 3726 + 3976 + 3805 = 11507 \
**1:** 800 + 881 + 852 = 2533 \
**2:** 214 + 293 + 255 = 762 \
**3:** 201 + 242 + 239 = 682 \
**7:** 2 + 1 + 7 = 10 \
**9:** 8 + 5 + 6 = 19 \
**.:** 664 + 526 + 571 = 1761

We have our **answer**: the 5.397605e-79 value represents a 0.

### Make a function

In [11]:
mental_health_df.columns

Index(['DPQ010', 'DPQ020', 'DPQ030', 'DPQ040', 'DPQ050', 'DPQ060', 'DPQ070',
       'DPQ080', 'DPQ090', 'DPQ100'],
      dtype='object')

In [12]:
# mh for mental health, I'm too lazy to type any more than that
mh_col = mental_health_df.columns
mh_col

## This was part of code that I used and then ended up deleting because it didn't work.
## Just if you were wondering why this useless block of code is here.

Index(['DPQ010', 'DPQ020', 'DPQ030', 'DPQ040', 'DPQ050', 'DPQ060', 'DPQ070',
       'DPQ080', 'DPQ090', 'DPQ100'],
      dtype='object')

In [13]:
def mh(row):
    if row['DPQ010'] == '\.':
        return 'missing'
    elif row['DPQ010'] == 1:
        return 'several days'
    elif row['DPQ010'] == 2:
        return 'more than half the days'
    elif row['DPQ010'] == 3:
        return 'nearly every day'
    elif row['DPQ010'] == 7:
        return 'refused'
    elif row['DPQ010'] == 9:
        return "don't know"
    else:
        return'not at all'


In [14]:
mental_health_df['depression_level'] = mental_health_df.apply(mh, axis=1)
mental_health_df[["DPQ010", "depression_level"]]

/Users/flowerbed/Desktop/School/9th Grade 2020 - 2021/2021 UCSF AI4ALL/GitHub/ai4all_nhanes/env/lib/python3.8/site-packages/pandas/core/frame.py:3607: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._set_item(key, value)


DPQ010 depression_level
0      5.397605e-79       not at all
4      1.000000e+00     several days
7      1.000000e+00     several days
8      5.397605e-79       not at all
9      5.397605e-79       not at all
...             ...              ...
28912  5.397605e-79       not at all
28915  5.397605e-79       not at all
28916  5.397605e-79       not at all
28917  5.397605e-79       not at all
28921  5.397605e-79       not at all

[15513 rows x 2 columns]

Then... we get to repeat for the other 9 questions!
There's probably a way to do it more effeciently with code but at this point it'd be more efficient for me to do it manually than to look up all the algorithms and documentation.

### Make some more functions

A handy dandy reminder of my ~~pain~~ work: ['DPQ010', 'DPQ020', 'DPQ030', 'DPQ040', 'DPQ050', 'DPQ060', 'DPQ070', 'DPQ080', 'DPQ090', 'DPQ100']

In [15]:
def mh(row):
    if row['DPQ020'] == '\.':
        return 'missing'
    elif row['DPQ020'] == 1:
        return 'several days'
    elif row['DPQ020'] == 2:
        return 'more than half the days'
    elif row['DPQ020'] == 3:
        return 'nearly every day'
    elif row['DPQ020'] == 7:
        return 'refused'
    elif row['DPQ020'] == 9:
        return "don't know"
    else:
        return'not at all'

mental_health_df['depression_level_2'] = mental_health_df.apply(mh, axis=1)
mental_health_df[["DPQ020", "depression_level_2"]]

/Users/flowerbed/Desktop/School/9th Grade 2020 - 2021/2021 UCSF AI4ALL/GitHub/ai4all_nhanes/env/lib/python3.8/site-packages/pandas/core/frame.py:3607: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._set_item(key, value)


DPQ020       depression_level_2
0      5.397605e-79               not at all
4      5.397605e-79               not at all
7      2.000000e+00  more than half the days
8      5.397605e-79               not at all
9      5.397605e-79               not at all
...             ...                      ...
28912  5.397605e-79               not at all
28915  1.000000e+00             several days
28916  5.397605e-79               not at all
28917  5.397605e-79               not at all
28921  5.397605e-79               not at all

[15513 rows x 2 columns]

In [16]:
def mh(row):
    if row['DPQ030'] == '\.':
        return 'missing'
    elif row['DPQ030'] == 1:
        return 'several days'
    elif row['DPQ030'] == 2:
        return 'more than half the days'
    elif row['DPQ030'] == 3:
        return 'nearly every day'
    elif row['DPQ030'] == 7:
        return 'refused'
    elif row['DPQ030'] == 9:
        return "don't know"
    else:
        return'not at all'

mental_health_df['depression_level_3'] = mental_health_df.apply(mh, axis=1)
mental_health_df[["DPQ030", "depression_level_3"]]

/Users/flowerbed/Desktop/School/9th Grade 2020 - 2021/2021 UCSF AI4ALL/GitHub/ai4all_nhanes/env/lib/python3.8/site-packages/pandas/core/frame.py:3607: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._set_item(key, value)


DPQ030 depression_level_3
0      5.397605e-79         not at all
4      1.000000e+00       several days
7      1.000000e+00       several days
8      5.397605e-79         not at all
9      5.397605e-79         not at all
...             ...                ...
28912  5.397605e-79         not at all
28915  5.397605e-79         not at all
28916  5.397605e-79         not at all
28917  5.397605e-79         not at all
28921  1.000000e+00       several days

[15513 rows x 2 columns]

In [17]:
def mh(row):
    if row['DPQ040'] == '\.':
        return 'missing'
    elif row['DPQ040'] == 1:
        return 'several days'
    elif row['DPQ040'] == 2:
        return 'more than half the days'
    elif row['DPQ040'] == 3:
        return 'nearly every day'
    elif row['DPQ040'] == 7:
        return 'refused'
    elif row['DPQ040'] == 9:
        return "don't know"
    else:
        return'not at all'

mental_health_df['depression_level_4'] = mental_health_df.apply(mh, axis=1)
mental_health_df[["DPQ040", "depression_level_4"]]

/Users/flowerbed/Desktop/School/9th Grade 2020 - 2021/2021 UCSF AI4ALL/GitHub/ai4all_nhanes/env/lib/python3.8/site-packages/pandas/core/frame.py:3607: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._set_item(key, value)


DPQ040 depression_level_4
0      5.397605e-79         not at all
4      1.000000e+00       several days
7      1.000000e+00       several days
8      5.397605e-79         not at all
9      5.397605e-79         not at all
...             ...                ...
28912  5.397605e-79         not at all
28915  5.397605e-79         not at all
28916  5.397605e-79         not at all
28917  5.397605e-79         not at all
28921  1.000000e+00       several days

[15513 rows x 2 columns]

In [18]:
def mh(row):
    if row['DPQ050'] == '\.':
        return 'missing'
    elif row['DPQ050'] == 1:
        return 'several days'
    elif row['DPQ050'] == 2:
        return 'more than half the days'
    elif row['DPQ050'] == 3:
        return 'nearly every day'
    elif row['DPQ050'] == 7:
        return 'refused'
    elif row['DPQ050'] == 9:
        return "don't know"
    else:
        return'not at all'

mental_health_df['depression_level_5'] = mental_health_df.apply(mh, axis=1)
mental_health_df[["DPQ050", "depression_level_5"]]

/Users/flowerbed/Desktop/School/9th Grade 2020 - 2021/2021 UCSF AI4ALL/GitHub/ai4all_nhanes/env/lib/python3.8/site-packages/pandas/core/frame.py:3607: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._set_item(key, value)


DPQ050 depression_level_5
0      5.397605e-79         not at all
4      1.000000e+00       several days
7      1.000000e+00       several days
8      5.397605e-79         not at all
9      5.397605e-79         not at all
...             ...                ...
28912  5.397605e-79         not at all
28915  5.397605e-79         not at all
28916  5.397605e-79         not at all
28917  5.397605e-79         not at all
28921  5.397605e-79         not at all

[15513 rows x 2 columns]

In [19]:
def mh(row):
    if row['DPQ060'] == '\.':
        return 'missing'
    elif row['DPQ060'] == 1:
        return 'several days'
    elif row['DPQ060'] == 2:
        return 'more than half the days'
    elif row['DPQ060'] == 3:
        return 'nearly every day'
    elif row['DPQ060'] == 7:
        return 'refused'
    elif row['DPQ060'] == 9:
        return "don't know"
    else:
        return'not at all'

mental_health_df['depression_level_6'] = mental_health_df.apply(mh, axis=1)
mental_health_df[["DPQ060", "depression_level_6"]]

/Users/flowerbed/Desktop/School/9th Grade 2020 - 2021/2021 UCSF AI4ALL/GitHub/ai4all_nhanes/env/lib/python3.8/site-packages/pandas/core/frame.py:3607: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._set_item(key, value)


DPQ060       depression_level_6
0      5.397605e-79               not at all
4      2.000000e+00  more than half the days
7      5.397605e-79               not at all
8      5.397605e-79               not at all
9      5.397605e-79               not at all
...             ...                      ...
28912  5.397605e-79               not at all
28915  5.397605e-79               not at all
28916  5.397605e-79               not at all
28917  5.397605e-79               not at all
28921  5.397605e-79               not at all

[15513 rows x 2 columns]

In [20]:
def mh(row):
    if row['DPQ070'] == '\.':
        return 'missing'
    elif row['DPQ070'] == 1:
        return 'several days'
    elif row['DPQ070'] == 2:
        return 'more than half the days'
    elif row['DPQ070'] == 3:
        return 'nearly every day'
    elif row['DPQ070'] == 7:
        return 'refused'
    elif row['DPQ070'] == 9:
        return "don't know"
    else:
        return'not at all'

mental_health_df['depression_level_7'] = mental_health_df.apply(mh, axis=1)
mental_health_df[["DPQ070", "depression_level_7"]]

/Users/flowerbed/Desktop/School/9th Grade 2020 - 2021/2021 UCSF AI4ALL/GitHub/ai4all_nhanes/env/lib/python3.8/site-packages/pandas/core/frame.py:3607: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._set_item(key, value)


DPQ070 depression_level_7
0      5.397605e-79         not at all
4      5.397605e-79         not at all
7      5.397605e-79         not at all
8      5.397605e-79         not at all
9      5.397605e-79         not at all
...             ...                ...
28912  5.397605e-79         not at all
28915  5.397605e-79         not at all
28916  5.397605e-79         not at all
28917  5.397605e-79         not at all
28921  5.397605e-79         not at all

[15513 rows x 2 columns]

In [21]:
def mh(row):
    if row['DPQ080'] == '\.':
        return 'missing'
    elif row['DPQ080'] == 1:
        return 'several days'
    elif row['DPQ080'] == 2:
        return 'more than half the days'
    elif row['DPQ080'] == 3:
        return 'nearly every day'
    elif row['DPQ080'] == 7:
        return 'refused'
    elif row['DPQ080'] == 9:
        return "don't know"
    else:
        return'not at all'

mental_health_df['depression_level_8'] = mental_health_df.apply(mh, axis=1)
mental_health_df[["DPQ080", "depression_level_8"]]

/Users/flowerbed/Desktop/School/9th Grade 2020 - 2021/2021 UCSF AI4ALL/GitHub/ai4all_nhanes/env/lib/python3.8/site-packages/pandas/core/frame.py:3607: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._set_item(key, value)


DPQ080 depression_level_8
0      5.397605e-79         not at all
4      1.000000e+00       several days
7      5.397605e-79         not at all
8      5.397605e-79         not at all
9      5.397605e-79         not at all
...             ...                ...
28912  5.397605e-79         not at all
28915  5.397605e-79         not at all
28916  5.397605e-79         not at all
28917  5.397605e-79         not at all
28921  5.397605e-79         not at all

[15513 rows x 2 columns]

In [22]:
def mh(row):
    if row['DPQ090'] == '\.':
        return 'missing'
    elif row['DPQ090'] == 1:
        return 'several days'
    elif row['DPQ090'] == 2:
        return 'more than half the days'
    elif row['DPQ090'] == 3:
        return 'nearly every day'
    elif row['DPQ090'] == 7:
        return 'refused'
    elif row['DPQ090'] == 9:
        return "don't know"
    else:
        return'not at all'

mental_health_df['depression_level_9'] = mental_health_df.apply(mh, axis=1)
mental_health_df[["DPQ090", "depression_level_9"]]

/Users/flowerbed/Desktop/School/9th Grade 2020 - 2021/2021 UCSF AI4ALL/GitHub/ai4all_nhanes/env/lib/python3.8/site-packages/pandas/core/frame.py:3607: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._set_item(key, value)


DPQ090 depression_level_9
0      5.397605e-79         not at all
4      5.397605e-79         not at all
7      5.397605e-79         not at all
8      5.397605e-79         not at all
9      5.397605e-79         not at all
...             ...                ...
28912  5.397605e-79         not at all
28915  5.397605e-79         not at all
28916  5.397605e-79         not at all
28917  5.397605e-79         not at all
28921  5.397605e-79         not at all

[15513 rows x 2 columns]

### Make a final function (for now)
This one gets its own section because the final milestone is so important for my sanity.

In [23]:
def mh(row):
    if row['DPQ100'] == '\.':
        return 'missing'
    elif row['DPQ100'] == 1:
        return 'several days'
    elif row['DPQ100'] == 2:
        return 'more than half the days'
    elif row['DPQ100'] == 3:
        return 'nearly every day'
    elif row['DPQ100'] == 7:
        return 'refused'
    elif row['DPQ100'] == 9:
        return "don't know"
    else:
        return'not at all'


In [24]:
mental_health_df['depression_level_10'] = mental_health_df.apply(mh, axis=1)
mental_health_df[["DPQ100", "depression_level_10"]]

/Users/flowerbed/Desktop/School/9th Grade 2020 - 2021/2021 UCSF AI4ALL/GitHub/ai4all_nhanes/env/lib/python3.8/site-packages/pandas/core/frame.py:3607: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._set_item(key, value)


DPQ100      depression_level_10
0               NaN               not at all
4      5.397605e-79               not at all
7      1.000000e+00             several days
8               NaN               not at all
9               NaN               not at all
...             ...                      ...
28912           NaN               not at all
28915  2.000000e+00  more than half the days
28916           NaN               not at all
28917           NaN               not at all
28921  5.397605e-79               not at all

[15513 rows x 2 columns]

## So, now what?
Technically I could visualize (I was planning to make some handy histograms) but I'll leave that to you all because 
1. I'm out of time,
2. I'm having issues with matplotlib, and without matplotlib I can't use seaborn, and the only thing I'm familiar with is seaborn, 
3. and yes, I know, but I don't really want to spend another 30 minutes looking up documentation and follow the visualization tutorial using the other thing.

## Reference links:
(basically everything I had to search up besides reference material I had from previous code) \
\
**Related to the project:** \
mental_health doc - https://wwwn.cdc.gov/Nchs/Nhanes/2011-2012/DPQ_G.htm#Data_Processing_and_Editing \
`.loc` - https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html \
`.dropna()` - https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.dropna.html
\
\
**Not *as* related to the project:**\
linking urls to text in md - https://docs.microsoft.com/en-us/contribute/how-to-write-links \
bolding text in md - https://www.markdownguide.org/basic-syntax/ \
strikethroughs in md - https://www.ionos.com/digitalguide/websites/web-development/markdown/

## A Final Note:
Alright, everyone. I worked hard with my single remaining braincell to make this \*aesthetic\* and understandable tomorrow morning, ~~(hey I just figured out that escaping with backslashes works in md too- can't believe it took me so long)~~ so please take your time and apprecieate the (minimal and monotone) \*aesthetics\* of this notebook. \
\
Thank you. \
~~--You know, should I have used $\LaTeX$ for everything instead?~~ \
~~---hey wait, strikethrough doesn't work on LaTeX???  \
----Well, LaTeX probably has its own strikethrough. And like LaTeX would allow the crossing out of their own name.~~

In [26]:
# little block of code so I don't have to work in this notebook anymore
mental_health_df.to_csv('mental_health_df.csv', index=False)